dcrlibwallet.go

```diff
@@ -2,70 +2,58 @@ package dcrlibwallet
 
 import (
 	"context"
-	"fmt"
-	"os"
 	"path/filepath"
 
-	"github.com/asdine/storm"
-	"github.com/decred/dcrwallet/errors"
-	"github.com/decred/dcrwallet/netparams"
-	"github.com/decred/dcrwallet/wallet"
-	"github.com/decred/dcrwallet/wallet/txrules"
+	"github.com/decred/dcrwallet/errors/v2"
+	"github.com/decred/dcrwallet/wallet/v3"
+	"github.com/decred/dcrwallet/wallet/v3/txrules"
+	"github.com/raedahgroup/dcrlibwallet/internal/loader"
+	"github.com/raedahgroup/dcrlibwallet/internal/netparams"
 	"github.com/raedahgroup/dcrlibwallet/txindex"
 	"github.com/raedahgroup/dcrlibwallet/utils"
-	"go.etcd.io/bbolt"
 )
 
-const logFileName = "dcrlibwallet.log"
+type Properties struct {
+	ID                     int    `storm:"id,increment"`
+	Name                   string `storm:"unique"`
+	DataDir                string
+	Seed                   string
+	SpendingPassphraseType int32
+	DiscoveredAccounts     bool
+}
 
 type LibWallet struct {
-	walletDataDir string
-	activeNet     *netparams.Params
-	walletLoader  *WalletLoader
-	wallet        *wallet.Wallet
-	txDB          *txindex.DB
-	configDB      *storm.DB
-	*syncData
+	*Properties `storm:"inline"`
+
+	activeNet    *netparams.Params
+	walletLoader *loader.Loader
+	wallet       *wallet.Wallet
+	txDB         *txindex.DB
+
+	synced     bool
+	syncing    bool
+	waiting    bool
+	rescanning bool
 
 	shuttingDown chan bool
 	cancelFuncs  []context.CancelFunc
 }
 
-func NewLibWallet(appDataDir, walletDbDriver string, netType string) (*LibWallet, error) {
-	// initialize logger before returning any error
-	errors.Separator = ":: "
-	initLogRotator(filepath.Join(appDataDir, logFileName))
+func NewLibWallet(props *Properties, walletDbDriver, netType string) (*LibWallet, error) {
```

Comment:
`NetType` should be part of the `Properties` properties. _(That should highlight why `Properties` is obviously not a good type name)_